File: backend/internal/domain/store.go
package domain

import "time"

// Store represents a physical or virtual point-of-sale location.
type Store struct {
	ID                     string     `db:"id"`
	Name                   string     `db:"name"`
	Address                string     `db:"address"`
	Phone                  string     `db:"phone"`
	TaxNumber              string     `db:"tax_number"`
	Currency               string     `db:"currency"`
	StoreType              string     `db:"store_type"` // retail | restaurant
	DefaultTaxPercentage   float64    `db:"default_tax_percentage"`
	LoyaltyPointsPerRupiah float64    `db:"loyalty_points_per_rupiah"`
	IsActive               bool       `db:"is_active"`
	CreatedAt              time.Time  `db:"created_at"`
	UpdatedAt              time.Time  `db:"updated_at"`
	DeletedAt              *time.Time `db:"deleted_at"`
}

File: frontend/src/types/store.ts
export interface Store {
  id: string;
  name: string;
  address: string;
  phone: string;
  tax_number: string;
  currency: string;
  store_type: 'retail' | 'restaurant';
  default_tax_percentage: number;
  is_active: boolean;
  created_at: string;
  updated_at: string;
  deleted_at?: string;
}
export interface StoreMember {
  user_id: string;
  user_name: string;
  user_email: string;
  role_id: string;
  role_name: string;
  is_active: boolean;
  joined_at: string;
}
